### DISCUSSION TITLE
Start tracing an application directly from the main method itself

### DISCUSSION THREAD

**Discussion by kushal-ti** [open-telemetry/opentelemetry-java-instrumentation#6238] [Q&A]
Based on what i have observed, mormally, tracing starts in a web server, running some kind of framework like spring, tracing starts from an incoming or outgoing request (either http request or a database call, etc).
I have an application which periodically schedules multiple containers of the same console app, all of whom push their logs to the same sink. I need a way to separate out the logs by which container produced them.
Instrumenting the logs with an individual trace/span id will accomplish this, but for that i need the agent to start instrumenting from the main of the application.

I know i can just add manual instrumentation in the code with the otel SDK, but that's an alternative i'd like to avoid if possible. It's a legacy project and making changes to the dependencies might break things in unexpected ways.

**Comment by kushal-ti** [comment]
I should also note that, i specifically want trace ids because each console app also makes some outbound HTTP requests to other services, and i would like it if the trace id that is generated and propagated in that outbound request could also be observed in the console app logs.

**Comment by laurit** [comment]
I'd urge you to give SDK a try, it is intended exactly for use cases like this. I find it unlikely that the agent will ever be automatically handle such custom needs.

**Comment by trask** [comment]
linking in duplicate https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/6238
